library/core/class.controller.php

```diff
@@ -1074,6 +1074,28 @@ public function json($Key, $Value = null) {
     }
 
     /**
+     * Encode a value as JSON or throw an exception on error.
+     *
+     * @param mixed $value
+     * @param int|null $options
+     * @return string
+     * @throws Gdn_UserException
+     */
+    private function jsonEncode($value, $options = null) {
+        if ($options === null) {
+            $options = JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES;
+        }
+
+        $encoded = json_encode($value, $options);
+
+        if (json_last_error() !== JSON_ERROR_NONE) {
+            throw new Gdn_UserException('A JSON encoding error has occurred');
```

Comment:
- This probably isn't a Gdn_UserException(). It's more of a fatal Exception in nature.
- You are missing an HTTP compatible error code (probably a 5xx error).
- We also usually make messages sentences with periods.
- You can conditionally show a more useful message on newer versions of PHP.